cloud/google/gce.py

```diff
@@ -423,7 +378,91 @@ def terminate_instances(module, gce, instance_names, zone_name):
             changed = True
 
     return (changed, terminated_instance_names)
+# Try to convert the user's metadata value into the format expected
+# by GCE.  First try to ensure user has proper quoting of a
+# dictionary-like syntax using 'literal_eval', then convert the python
+# dict into a python list of 'key' / 'value' dicts.  Should end up
+# with:
+# [ {'key': key1, 'value': value1}, {'key': key2, 'value': value2}, ...]
+def cleanup_metadata(metadata):
+    if metadata:
+        try:
+            md = literal_eval(metadata)
+            if not isinstance(md, dict):
+                raise ValueError('metadata must be a dict')
+        except ValueError, e:
+            # TODO: use module.fail_json instead of print here
+            print("failed=True msg='bad metadata: %s'" % str(e))
+            sys.exit(1)
+        except SyntaxError, e:
+            # TODO: use module.fail_json instead of print here
+            print("failed=True msg='bad metadata syntax'")
+            sys.exit(1)
+
+        items = []
+        for k,v in md.items():
+            items.append({"key": k,"value": v})
 
+        metadata = {'items': items}\
+    return metadata
+
+def create_instance_by_name(module, gce, name):
+    # TODO: implement using create_node
+    pass
+
+def create_instances_by_names(module, gce):
+    instance_json_data = []
+    changed = False
+    for name in module.params['instance_names']:
+        is_changed, inst_json = create_instance_by_name(module, gce, name)
+        instance_json_data.append(inst_json)
+        if is_changed:
+            changed = is_changed
+    return (changed, instance_json_data, [ i['name'] for i in instance_json_data ])
+
+def create_instances_by_count(module, gce):
+    # TODO: implement using ex_create_multiple_nodes
```

Comment:
That underlying libcloud function may need some attention.  I don't think I've been doing a stellar job at keeping it updated.